centreon/www/class/centreonDowntime.class.php

```diff
@@ -878,50 +918,68 @@ public function deletePeriods($id)
+                'INSERT INTO downtime_hostgroup_relation (dt_id, hg_hg_id) VALUES (:id, :obj_id)'
+            ),
+            self::TYPE_SERVICE => $this->db->prepare(<<<'SQL'
+                INSERT INTO downtime_service_relation (dt_id, host_host_id, service_service_id)
+                VALUES (:id, :host_id, :service_id)
+                SQL
+            ),
+            self::TYPE_SERVICE_GROUP => $this->db->prepare(
+                'INSERT INTO downtime_servicegroup_relation (dt_id, sg_sg_id) VALUES (:id, :obj_id)'
+            ),
+            default => null
+        };
+
+        if ($statement === null) {
+            return;
         }
-        foreach ($obj_ids as $obj_id) {
-            if ($obj_type == 'svc') {
-                $obj_id = str_replace('-', ', ', $obj_id);
+
+        foreach ($objIds as $ids) {
+            if ($objType === self::TYPE_SERVICE) {
+                [$hostId, $serviceId] = explode('-', $ids);
+                $statement->bindValue(':host_id', $hostId, \PDO::PARAM_INT);
+                $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
+            } else {
+                $statement->bindValue(':obj_id', $ids, \PDO::PARAM_INT);
             }
-            $queryInsert = str_replace('%obj_id%', $obj_id, $query);
-            $this->db->query($queryInsert);
+            $statement->bindParam(':id', $id, \PDO::PARAM_INT);
+            $statement->execute();
         }
     }
 
     /**
-     * Delete all relations for a downtime
+     * Delete all downtime relations
      *
      * @param int $id The downtime id
      */
-    public function deteleRelations($id)
+    public function deleteRelations(int $id): void
     {
-        $query = "DELETE FROM downtime_host_relation WHERE dt_id = " . $id;
-        $this->db->query($query);
-        $query = "DELETE FROM downtime_hostgroup_relation WHERE dt_id = " . $id;
-        $this->db->query($query);
-        $query = "DELETE FROM downtime_service_relation WHERE dt_id = " . $id;
-        $this->db->query($query);
-        $query = "DELETE FROM downtime_servicegroup_relation WHERE dt_id = " . $id;
-        $this->db->query($query);
+        $statement = $this->db->prepare('DELETE FROM downtime_host_relation WHERE dt_id = :id');
+        $statement->bindParam(':id', $id, \PDO::PARAM_INT);
+        $statement->execute();
+
+        $statement = $this->db->prepare('DELETE FROM downtime_hostgroup_relation WHERE dt_id = :id');
+        $statement->bindParam(':id', $id, \PDO::PARAM_INT);
+        $statement->execute();
+
+        $statement = $this->db->prepare('DELETE FROM downtime_service_relation WHERE dt_id = :id');
+        $statement->bindParam(':id', $id, \PDO::PARAM_INT);
+        $statement->execute();
+
+        $statement = $this->db->prepare('DELETE FROM downtime_servicegroup_relation WHERE dt_id = :id');
+        $statement->bindParam(':id', $id, \PDO::PARAM_INT);
+        $statement->execute();
     }
```

Comment:
I know the behaviour was like this before, but those queries shouldn't be in a transaction ?